### PR TITLE
Adding PGMONETA_VERSION in backup.info Issue #235

### PR DESCRIPTION
**Pull request by palak-chaturvedi**
The backup.info can updated to add version number to it from the pgmoneta.h file.

example
VERSION = 0.10.0

### CODE REVIEW

**Comment by jesperpedersen** [comment]
You need to add yourself to the `AUTHORS` file.

Then squash your commits, rebase and force push.

https://www.git-tower.com/learn/git/faq/git-squash
https://github.com/progit/progit2/releases/tag/2.1.423

**Comment by Jubilee101** [comment]
Hi there, I think as for the label name we want `PGMONETA` instead of `PGMONETA_VERSION`.

And you need to fix your commit message, we want something like `[#235]your commit message`.

**Comment by jesperpedersen** [comment]
I'm ok with `PGMONETA_VERSION`

**Comment by Jubilee101** [comment]
Hi, seems that you only need to change the commit message then. Change it and force push. Feel free to let us know if you have problems with Git and stuff, it's completely normal :)

**Comment by palak-chaturvedi** [comment]
I am able to use git rebase but there is someone's else commit shown in my git log along with mine.
image

**Comment by palak-chaturvedi** [comment]
My file changes are only in info.c 
It seems I might have merged one commit in mine.
Do I need to recreate a new PR because I have rebased forced push in this branch?

**Comment by jesperpedersen** [comment]
It looks like you didn't rebase properly... It is easy with tools like gitk to verify that you have a single stream of commits.

You don't need to create a new pull request, just force push your branch and it should be sorted.

**Comment by Jubilee101** [comment]
Hi Palak, I don't think you are rebasing correctly. If you want to rebase and catch up with main branch, with your commits on top of current main branch. You should do the following:
1. Drop the 3 commits after your update info.c commit. This could be done by a single rebase - run `git rebase -i HEAD~4`, you should see a config file opened with vim with 4 lines, each with a `pick` in front of the commit message. If you want to drop the commit, simply change `pick` to `drop`. After you are done with the config file, save it and exit - you probably need to know basic vim command.
2.  Add your name to AUTHOR.md again, and commit. This time you need to squash this commit into your previous update info commit. Do `git rebase -i HEAD~2`. Change `pick` of the later commit, in your case the update author commit, into `fix`, so that the commit is fixed up. You should also change your commit message, as we talked about, morden IDE offers that function, but you can also do that in rebase, to do that, apart from previous modification, change `pick` of your update info commit into `reword`, and save & exit. It will bring you to a new vim file where you could edit the commit message, edit and save.
3. Now you have your changes squashed in one commit. You should rebase it to let it sit on top of the lastest commit in the remote repo. To do that, first checkout to main branch and pull the latest change from the upstream. Then go back to your dev branch, do `git rebase main`. Then you'll see your changes already rebased to top. There's tiny chance that you might experience conflict, if that happens, find and resolve your conflict, meaning decide what changes should be kept or modified, do `git add -u` and do `git rebase --continue`.

After you are ready, do `git push --force` to push your commits to this PR.

I hope this is clear enough, there are also some tutorials you can find on google or youtube. Feel free to explore around.

Let me know if you have any questions:) It's normal to get messed up with git at first, just get your hands dirty, try to get it resolved, and understand why. You'll get there:)

**Comment by jesperpedersen** [comment]
Hmm, we should add this to the Prometheus interface too...

```
#TYPE pgmoneta_version gauge
```

So

```
pgmoneta_version{version="0.11.0"} 1
```

after the `pgmoneta_state` definition... Remember the documentation on `home_page()` too

**Comment by jesperpedersen** [comment]
@palak-chaturvedi Remember to resolve the conflict - should be simple to do

**Comment by jesperpedersen** [comment]
https://github.com/pgmoneta/pgmoneta/blob/main/doc/DEVELOPERS.md#basic-git-guide

File: src/libpgmoneta/prometheus.c [hunk 1]
```diff
@@ -949,7 +952,11 @@ general_information(int client_fd)
    data = pgmoneta_append(data, "pgmoneta_state ");
    data = pgmoneta_append(data, "1");
    data = pgmoneta_append(data, "\n\n");
-
+   data = pgmoneta_append(data, "#HELP pgmoneta_version The version of pgmoneta\n");
+   data = pgmoneta_append(data, "#TYPE pgmoneta_version gauge\n");
+   data = pgmoneta_append(data, "pgmoneta_version ");
```
Comment (jesperpedersen) [inline src/libpgmoneta/prometheus.c:957]: You have to do this as

```
pgmoneta_version{version="0.11.0"} 1
```

Check it in Prometheus / Grafana

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by jesperpedersen** [comment]
@palak-chaturvedi How is this going ?

File: src/libpgmoneta/prometheus.c (hunk 1 above)
Comment (palak-chaturvedi) [reply src/libpgmoneta/prometheus.c:957]: what is 1 supposed to mean?
and am I not supposed to add this #TYPE code 
or the `pgmoneta_version{version="0.11.0"} 1` will be written here only?


Sorry for late reply. I had my farewell going on in my college. So could not get back.

File: src/libpgmoneta/prometheus.c [hunk 2]
```diff
@@ -949,7 +952,12 @@ general_information(int client_fd)
    data = pgmoneta_append(data, "pgmoneta_state ");
    data = pgmoneta_append(data, "1");
    data = pgmoneta_append(data, "\n\n");
-
+   data = pgmoneta_append(data, "#HELP pgmoneta_version The version of pgmoneta\n");
+   data = pgmoneta_append(data, "#TYPE pgmoneta_version gauge\n");
+   data = pgmoneta_append(data, "pgmoneta_version{version=' ");
```
Comment (jesperpedersen) [inline src/libpgmoneta/prometheus.c:957]: Single `'`

File: src/libpgmoneta/prometheus.c [hunk 3]
```diff
@@ -949,7 +952,12 @@ general_information(int client_fd)
...
+   data = pgmoneta_append(data, VERSION);
+   data = pgmoneta_append(data, "'} 1");
```
Comment (jesperpedersen) [inline src/libpgmoneta/prometheus.c:959]: Single `'`

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: src/libpgmoneta/prometheus.c (hunk 1 above)
Comment (jesperpedersen) [reply src/libpgmoneta/prometheus.c]: We need the `1` because that is how Prometheus works

**Comment by jesperpedersen** [comment]
You are very close

**Comment by jesperpedersen** [comment]
Merged.

Thanks for your contribution !

**Pull request merged**